modules/local/filter_ssu.nf

```diff
@@ -0,0 +1,74 @@
+process FILTER_SSU {
+    tag "${fasta}"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconductor::biostrings=2.58.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bioconductor-biostrings:2.58.0--r40h037d062_0' :
+        'quay.io/biocontainers/bioconductor-biostrings:2.58.0--r40h037d062_0' }"
+
+    input:
+    path(fasta)
+    path(table)
+    path(lists)
+
+    output:
+    path( "stats.ssu.tsv" )      , emit: stats
+    path( "ASV_table.ssu.tsv" )  , emit: asv
+    path( "ASV_seqs.ssu.fasta" ) , emit: fasta
+    path "versions.yml"          , emit: versions
+
+    script:
+    def kingdom = params.filter_ssu ?: "bac,arc,mito,euk"
+    """
+    #!/usr/bin/env Rscript
+
+    #load packages
+    suppressPackageStartupMessages(library(Biostrings))
+
+    #use only selected kingdom
+    dir.create("./selection")
+    kingdom <- as.list(strsplit("$kingdom", ",")[[1]])
+    for (x in kingdom) {
```

Comment:
Probably not, but it does make the job. Also only max 4 iterations, so no speed concerns. This part is supposed to collect the files that are requested instead of taking all.

Reply:
👍